taxi/views.py

```diff
@@ -12,4 +13,38 @@ def index(request):
         "num_manufacturers": Manufacturer.objects.count(),
     }
 
-    return render(request, "taxi/index.html", context=context)
+    return render(request,
+                  "taxi/index.html", context=context)
+
+
+class ManufacturerListView(generic.ListView):
+    model = Manufacturer
+    queryset = Manufacturer.objects.all().order_by("name")
+    paginate_by = 5
+    template_name = "taxi/manufacturer_list.html"
+
+
+class CarListView(generic.ListView):
+    model = Car
+    queryset = Car.objects.select_related("manufacturer")
+    paginate_by = 5
+    template_name = "taxi/car_list.html"
+
+
+class CarDetailView(generic.DetailView):
+    model = Car
+    template_name = "taxi/car_detail.html"
+    queryset = (Car.objects.
+                prefetch_related("manufacturer", "drivers"))
```

Comment:
```suggestion
                select_related("manufacturer").prefetch_related("drivers"))
```
manufacturer - one
drivers - many